libpoolprotocols/PoolManager.cpp

```diff
@@ -316,11 +316,17 @@ void PoolManager::setActiveConnection(unsigned int idx)
     }
 }
 
-const URI* PoolManager::getActiveConnection()
+URI PoolManager::getActiveConnectionCopy()
 {
-    if (m_connections.size() > m_activeConnectionIdx)
-        return &m_connections.at(m_activeConnectionIdx);
-    return nullptr;
+    URI r = URI(":0");
+    try
+    {
+       r = m_connections.at(m_activeConnectionIdx);
```

Comment:
vectors are not thread safe. You could still end up with an invalid copy. For example: Say size = 5, and index = 3. The at() method only checks that index < size. What if at() is interrupted by a delete of index 2 by the ApiServer?

Reply:
The copy is needed, but you also need to surround all accesses to m_connections and m_connection_index with a lock, to make it thread safe.

Reply:
OK. I understand!
I've just moved the problem to an invisible (=std::vector)  .at() function where the same code [ if(vector.size()........] is done like I did it before.
... stay tuned ... next try will come ;-) ...